crates/cargo-contract/src/cmd/call.rs

```diff
@@ -74,7 +79,7 @@ pub struct CallCommand {
     proof_size: Option<u64>,
     /// The value to be transferred as part of the call.
     #[clap(name = "value", long, default_value = "0")]
-    value: BalanceVariant,
+    value: BalanceVariant<<DefaultEnvironment as Environment>::Balance>,
```

Comment:
We may need to use a general concrete type here and for other command fields which use types like this, because clap is responsible for instantiating the commands, and we want to be able (eventually) to choose which `Environment` impl to use for invoking the actual commands. So there will need to be a runtime conversion from the parsed input type to the chosen env type.

Reply:
Yes, This PR is just a first with focus on extrinisics crate.